Fix environment for protobuf compilation in grpc

Flamefire opened the pull request:
Add use_default_shell_env = True to protoc invocation for grpc to mirror what the protobuf cc_proto_library & co are doing
Fixes a failure in invocing protoc when it is build in a non-default environment (e.g. with a custom LD_LIBRARY_PATH)

Fixes #11852, fixes #11855

This is a regression from 2.x to 3.0. Not sure what causes these issues but likely the upgrade of the used protobuf from 3.6 to 3.11

c-mita commented:
The rule in question comes from github.com/grpc/grpc/blob/master/bazel/generate_cc.bzl and so might need fixing there.

Flamefire commented:
Good catch. I'm currently building TensorFlow with this applied and they call `cc_grpc_library` -> `generate_cc` with `load("@com_github_grpc_grpc//bazel:generate_cc.bzl", "generate_cc")` which I assume imports the file you mentioned and which has the same problem. Hence my build fails

meisterT commented:
What's the state of this PR?

Flamefire commented:
From my side this is still ready and we use this as a patch already. The Bazel guys didn't react on my upstream PR at https://github.com/grpc/grpc/pull/23664 yet

c-mita commented:
I think this'll be fine as is, but we need a new patch file to reflect the change (i.e. a new https://github.com/bazelbuild/bazel/blob/master/third_party/grpc/bazel_1.32.0.patch).

Can you do this?

Flamefire commented:
Rebased and updated the patch. Hope that is what you meant. A comment from you on https://github.com/grpc/grpc/pull/23664 so this gets upstreamed would be nice.

c-mita reviewed (approved):
LGTM

Flamefire commented:
Looks like only the 2nd commit was taken. Is that intentional?

meisterT commented:
@meteorcloudy can you check what went wrong, please?

meteorcloudy commented:
Oh, sorry! I forgot to squash them first, will fix this now!

meteorcloudy commented:
First commit is also in now: https://github.com/bazelbuild/bazel/commit/debbcdefe5f608cd27ffae3e4387f38dad3f664a
